src/simplicial-hott/06-2cat-of-segal-types.rzk.md

```diff
@@ -440,3 +501,186 @@ the "Gray interchanger" built from two commutative triangles.
+
+The fiber of postcomposition by a map $f: \prod_{t : I|\psi} A (t) \to B (t)$ is
+equivalent to the family of fibers of $f\_t$.
+
+```rzk
+#def postcomp-Π-ext
+  ( I : CUBE)
+  ( ψ : I → TOPE)
+  ( ϕ : ψ → TOPE)
+  ( A B : ψ → U)
+  ( a : (t : ϕ) → A t)
+  ( f : (t : ψ) → (A t → B t))
+  : ((t : ψ) → A t [ϕ t ↦ a t]) → ((t : ψ) → B t [ϕ t ↦ f t (a t)])
+  := ( \ α t → f t (α t))
+
+#def fiber-postcomp-Π-ext -- already defined as relative extension type in 03-extension-types
+  ( I : CUBE)
+  ( ψ : I → TOPE)
+  ( ϕ : ψ → TOPE)
+  ( A B : ψ → U)
+  ( a : (t : ϕ) → A t)
+  ( f : (t : ψ) → (A t → B t))
+  ( β : (t : ψ) → B t [ϕ t ↦ f t (a t)])
+  : U
+  :=
+  fib
+  ( (t : ψ) → A t [ϕ t ↦ a t])
+  ( (t : ψ) → B t [ϕ t ↦ f t (a t)])
+  ( postcomp-Π-ext I ψ ϕ A B a f)
+  ( β)
+
+#def fiber-family-ext
+  ( I : CUBE)
+  ( ψ : I → TOPE)
+  ( ϕ : ψ → TOPE)
+  ( A B : ψ → U)
+  ( a : (t : ϕ) → A t)
+  ( f : (t : ψ) → (A t → B t))
+  ( β : (t : ψ) → B t [ϕ t ↦ f t (a t)])
+  : U
+  := (t : ψ) → fib (A t) (B t) (f t) (β t) [ϕ t ↦ (a t, refl)]
+
+#def equiv-fiber-postcomp-Π-ext-fiber-family-ext
+  ( I : CUBE)
+  ( ψ : I → TOPE)
+  ( ϕ : ψ → TOPE)
+  ( A B : ψ → U)
+  ( a : (t : ϕ) → A t)
+  ( f : (t : ψ) → (A t → B t))
+  ( β : (t : ψ) → B t [ϕ t ↦ f t (a t)])
+  : Equiv
+    ( fiber-postcomp-Π-ext I ψ ϕ A B a f β)
+    ( fiber-family-ext I ψ ϕ A B a f β)
+  :=
+  equiv-comp
+  ( fiber-postcomp-Π-ext I ψ ϕ A B a f β)
+  ( relative-extension-type I ψ ϕ A B f a β)
+  ( fiber-family-ext I ψ ϕ A B a f β)
+  ( equiv-relative-extension-type-fib extext I ψ ϕ A B f a β)
+  ( inv-equiv-axiom-choice I ψ ϕ A (\ t x → f t x = β t) a (\ t → refl))
+```
+
+In particular, if $f: \prod_{t : I|\psi} A (t) \to B (t)$ is a family of
+equivalence then the fibers of postcomposition by f are contractible.
+
+```rzk
+#def is-contr-fiber-family-ext-contr-fib
+  ( I : CUBE)
+  ( ψ : I → TOPE)
+  ( ϕ : ψ → TOPE)
+  ( A B : ψ → U)
+  ( a : (t : ϕ) → A t)
+  ( f : (t : ψ) → (A t → B t))
+  ( β : (t : ψ) → B t [ϕ t ↦ f t (a t)])
+  ( family-equiv-f : (t : ψ) → is-equiv (A t) (B t) (f t))
+  : is-contr (fiber-family-ext I ψ ϕ A B a f β)
+  :=
+  weakextext I ψ ϕ
+  ( \ t → fib (A t) (B t) (f t) (β t))
+  ( \ t → is-contr-map-is-equiv (A t) (B t) (f t) (family-equiv-f t) (β t))
+  ( \ t → (a t, refl))
+
+#def is-contr-fiber-postcomp-Π-ext-is-equiv-fam uses (weakextext extext)
+  ( I : CUBE)
+  ( ψ : I → TOPE)
+  ( ϕ : ψ → TOPE)
+  ( A B : ψ → U)
+  ( a : (t : ϕ) → A t)
+  ( f : (t : ψ) → (A t → B t))
+  ( β : (t : ψ) → B t [ϕ t ↦ f t (a t)])
+  ( family-equiv-f : (t : ψ) → is-equiv (A t) (B t) (f t))
+  : is-contr (fiber-postcomp-Π-ext I ψ ϕ A B a f β)
+  :=
+  is-contr-equiv-is-contr'
+  ( fiber-postcomp-Π-ext I ψ ϕ A B a f β)
+  ( fiber-family-ext I ψ ϕ A B a f β)
+  ( equiv-fiber-postcomp-Π-ext-fiber-family-ext I ψ ϕ A B a f β)
+  ( is-contr-fiber-family-ext-contr-fib I ψ ϕ A B a f β family-equiv-f)
+
+#def is-equiv-postcomp-Π-ext-is-equiv uses (weakextext extext)
```

Comment:
I just noticed that this already exists in the library. It is called `is-equiv-extensions-is-equiv`. But actually I like your proof, because it gives more information about the fibers in the case where the map is not an equivalence. And also it does not require first proving the statement for `BOT` and then bootstrapping.

So maybe you could consider moving some of the material here to the section "Functoriality of extension types" and potentially replacing the other proof with this one?

Reply:
@StiephenPradal did you want to make these changes or would you mind if I did it for you? I think this is the only remaining discussion. Otherwise we're ready to merge.

Reply:
Sure, I'll make the changes now!

Reply:
I tried to make the changes as coherent as possible. Let me know if I should do something differently.

Reply:
This looks good to me. @TashiWalde is this what you had in mind?